src/main/java/com/conveyal/r5/streets/StreetRouter.java

```diff
@@ -229,7 +231,7 @@ public TIntIntMap getReachedStops() {
      * objective variable for the optimal path to that vertex.
      */
     public TIntIntMap getReachedVertices () {
-        TIntIntMap result = new TIntIntHashMap();
+        TIntIntMap result = new TIntIntHashMap(DEFAULT_CAPACITY, DEFAULT_LOAD_FACTOR, -1, Integer.MAX_VALUE);
```

Comment:
Javadoc for this method should mention that the returned map has a noEntryValue of Integer.MAX_VALUE and a noEntryKey of -1 independent of the objective variable.

As I was auditing the code, this constructor threw me off for a while because I misinterpreted the -1 as the noEntryValue, but it's actually the noEntryKey. I didn't expect Trove to have a configurable noEntryKey, and don't really understand why. Although -1 does seem more appropriate than zero, even after digging into the Trove source code I don't quite understand whether there are any side effects to setting noEntryKey to -1. 

Taking a step back, maybe a more robust solution would be for this method to return a more clearly defined type than TIntIntMap, such as CostToVertexFunction (which has Javadoc unambiguously stating that MAX_VALUE always means unreachable). Then TransitLayer#stopToVertexDistanceTables could also be more strongly typed as a List<CostToVertexFunction>, or better yet an even more strongly typed List<DistanceToVertex>, where DistanceToVertex extends CostToVertexFunction but doesn't add anything other than a method returning a RoutingVariable of DISTANCE_MILLIMETERS. Contrary to the TODO comment on CostToVertexFunction I don't think we'd want to wrap CostToVertexFunction to achieve this, but instead make it an abstract class with two subclasses (one returning DISTANCE_MILLIMETERS and the other DURATION_SECONDS from a getRoutingVariable method). CostToVertexFunction could even be made less functional, given a name like CostToVertex (or CostToVertexTable) and extend or contain a TIntIntMap, so StreetRouter or anything else could just directly instantiate a CostToVertex and add values to it, making it impossible to misconfigure any maps with wrong noEntryValues.

That may be too much to add to this PR though - we could try that later in another round of refactoring.